compiler/rustc_errors/Cargo.toml

```diff
@@ -10,9 +10,11 @@ derive_setters = "0.1.6"
 rustc_ast = { path = "../rustc_ast" }
 rustc_ast_pretty = { path = "../rustc_ast_pretty" }
 rustc_data_structures = { path = "../rustc_data_structures" }
+rustc_error_codes = { path = "../rustc_error_codes" }
```

Comment:
The point of `rustc_error_codes` was to avoid any dependency on the error code comments from the explanations this early in the dependency tree.
OTOH, the current way to decouple messages from code is using fluent. Should we go in this direction in the future? Or is the modification rate for this crate low enough now?

Reply:
Interesting... the top-level comment in the crate says "This library is used to gather all error codes into one place, the goal being to make their maintenance easier."  But nothing about the dependency graph. (I'm just noting this because it's a pet peeve of mine when that kind of important but non-obvious consideration isn't put into a comment somewhere, because it's exactly the kind of thing that should be put into a comment.)

Back to the original point: the approach we landed on of defining `Ennnn` constants makes the additional dependency unavoidable. Because we need one constant per error code, and those error codes are used everywhere, and they depend on the error code list. With the `E!(0123)` macro approach this additional dependency could have been avoided, because `ErrCode` and `E!` could have been defined in `rustc_errors`.

I see 93 commits to `rustc_error_codes` in 2023, though about a quarter of those appear to be merge commits. That doesn't seem terribly high to me.